pycbc/inject/inject.py

```diff
@@ -397,52 +670,101 @@ def make_strain_from_inj_object(self, inj, delta_t, detector_name,
             Sample rate to make injection at.
         detector_name : string
             Name of the detector used for projecting injections.
-        f_lower : {None, float}, optional
-            Low-frequency cutoff for injected signals. If None, use value
-            provided by each injection.
-        distance_scale: {1, float}, optional
-            Factor to scale the distance of an injection with. The default is
-            no scaling.
+        distance_scale: float, optional
+            Factor to scale the distance of an injection with. The default (=1)
+            is no scaling.
 
         Returns
         --------
         signal : float
             h(t) corresponding to the injection.
         """
         detector = Detector(detector_name)
-        if f_lower is None:
-            f_l = inj.f_lower
-        else:
-            f_l = f_lower
-
-        name, phase_order = legacy_approximant_name(inj.waveform)
 
         # compute the waveform time series
-        hp, hc = get_td_waveform(
-            inj, approximant=name, delta_t=delta_t,
-            phase_order=phase_order,
-            f_lower=f_l, distance=inj.distance,
-            **self.extra_args)
-
-        hp /= distance_scale
-        hc /= distance_scale
+        hp, hc = ringdown_td_approximants[inj['approximant']](
+            inj, delta_t=delta_t, **self.extra_args)
 
-        hp._epoch += inj.get_time_geocent()
-        hc._epoch += inj.get_time_geocent()
+        hp._epoch += inj['tc']
+        hc._epoch += inj['tc']
 
-        # taper the polarizations
-        hp_tapered = wfutils.taper_timeseries(hp, inj.taper)
-        hc_tapered = wfutils.taper_timeseries(hc, inj.taper)
+        if distance_scale != 1:
```

Comment:
This distance_scale comes from the command options, right? It is not related to the "distance" parameter in the injection file, is that correct?

Reply:
Yeah, that's right. I only added it here because the strain module expects to be able to pass a distance scale option to the apply function, which I think it gets from `--injection-scale-factor`. We don't really use it for anything.

Reply:
That's what I thought. I know it was there before, but at that time we didn't have a distance parameter in the ringdown approximant so I didn't bother. Just wanted to be 100% sure we wouldn't be scaling by the distance twice. Thanks.

Reply:
This is used for sanity checks. For example make the injection upside down to check if you have an unfavourable noise realization.